versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/DatabaseAdapterConfig.java

```diff
@@ -111,6 +112,38 @@ default int getDefaultMaxKeyListSize() {
     return DEFAULT_MAX_KEY_LIST_SIZE;
   }
 
+  /**
+   * Maximum size of a database object/row.
+   *
+   * <p>This parameter is respected for {@link
+   * org.projectnessie.versioned.persist.adapter.KeyListEntity}.
+   *
+   * <p>Not all kinds of databases have hard limits on the maximum size of a database object/row.
+   *
+   * <p>This value must not be "on the edge" - means: it must leave enough room for
+   * database-serialization overhead and similar.
+   *
+   * <p>Values {@code <=0} are illegal, defaults to {@link #getDefaultMaxKeyListEntitySize()}.
+   */
+  @Value.Default
+  default int getMaxKeyListEntitySize() {
+    return getDefaultMaxKeyListEntitySize();
+  }
+
+  /**
+   * Database adapter implementations that actually do have a hard technical or highly recommended
+   * limit on a maximum db-object / db-row size limitation should override this method and return a
+   * "good" value.
+   *
+   * <p>As for {@link #getMaxKeyListSize()}, this value must not be "on the edge" - means: it must
+   * leave enough room for database-serialization overhead * and similar.
+   *
+   * <p>Defaults to {@value #DEFAULT_MAX_KEY_LIST_ENTITY_SIZE}.
+   */
+  default int getDefaultMaxKeyListEntitySize() {
```

Comment:
I'm not sure how it is going to work :thinking: In current config usage the `getMaxKeyListEntitySize` setting  apparently always overrides `getDefaultMaxKeyListEntitySize`... Is it worth having both of these settings? If we want the "default" size to be database-specific, should the code make sure `getDefaultMaxKeyListEntitySize` takes precedence?

Reply:
Should the method then be on the `DatabaseAdapter` interface? Or perhaps of the adapter factory?

Reply:
right now this value comes from the same source (e.g. `QuarkusVersionStoreAdvancedConfig`) as the other "key list" size, which makes it a bit confusing from my POV :thinking:

Reply:
(See below - will do in a separate PR)

Reply:
But yes, it  (the hard limit) is an implementation concern and should be handled there.